### ISSUE TITLE
Frequent peer removal and inserting

### ISSUE THREAD

**Issue by ManuGowda** [LiskHQ/lisk#2455]
### Expected behavior
Peers should not be removed frequently
### Actual behavior
when a `1.1.0-rc.0` peer is connected with the testnet network, the `1.1.0-rc.0` node removes and add the peer frequently
```
[dbg] 2018-10-09 10:04:18 | Removed peer -
[inf] 2018-10-09 10:10:06 | INSERT peer -
```
since time `2018-10-09 08:59:00` to `2018-10-09 10:12:35` peers has been removed and inserted `8699` times
### Steps to reproduce
logs are available at `206.189.1.129`
### Which version(s) does this affect? (Environment, OS, etc...)
`1.1.0`

**Comment by ManuGowda** [comment]
also getting log of `Unable to match an address to the peer`
```
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"83.136.254.104:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"95.179.141.55:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Successfully notified peer about self - {"peer":"81.169.184.136:7001"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"80.240.27.196:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"178.62.106.202:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"185.183.156.148:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"51.15.201.186:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"167.99.217.38:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"178.128.248.109:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Successfully notified peer about self - {"peer":"95.216.137.205:7001"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"198.50.245.220:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Successfully notified peer about self - {"peer":"144.217.16.114:7001"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"195.201.150.178:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Successfully notified peer about self - {"peer":"192.154.224.141:7001"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"104.251.212.43:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"209.97.141.55:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Successfully notified peer about self - {"peer":"104.236.152.240:7001"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"94.237.64.33:7001","err":"Error: Unable to match an address to the peer"}
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"142.44.247.234:7001","err":"Error: Unable to match an address to the peer"}
[inf] 2018-10-09 10:21:00 | Releasing enqueued broadcasts
[inf] 2018-10-09 10:21:00 | Queue empty
[dbg] 2018-10-09 10:21:00 | Transport->broadcastHeaders: Failed to notify peer about self - {"peer":"149.28.175.89:7001","err":"Error: Unable to match an address to the peer"}
```

**Comment by jondubois** [comment]
Changing the version number from `1.1.0-rc.0` to `1.1.0` in `package.json` fixes the issue. Removing the version suffix causes the number of disconnect errors to drop dramatically.

The problem is that old `1.0.0-rc.x` nodes cannot handle the full version number of our new nodes like `1.1.0-rc.0`; we made a fix before which sanitises our version number to remove the `-rc.x` suffix, it doesn't appear to fully resolve the issue. It's possible that old `1.0.0-rc.x` nodes are getting our suffixed version number from some other way (maybe by requesting our peer list).

I tried to sanitize the version numbers in our new node's peer list but it doesn't seem to resolve the issue. There may be other places where our real suffixed version number is being passed.

Rather than adding hacks in the code to get the suffixed versioning to work, we should consider a different tagging/versioning/deployment approach for the next release.
